ambari-server/src/main/resources/stacks/service_advisor.py

```diff
@@ -115,6 +124,59 @@ def getServiceComponentLayoutValidations(self, services, hosts):
     """
     return []
 
+  def getServiceComponentCardinalityValidations(self, services, hosts, service_name):
```

Comment:
Missing doc, which would be useful for a function like this.

Reply:
fixed